custom_components/tuya_local/devices/momcozy_white_noise_machine.yaml

```diff
@@ -0,0 +1,278 @@
+      - id: 9
+        name: volume_level
+        type: integer
+        range:
+          min: 10
+          max: 100
+        mapping:
+          - scale: 100
+      - id: 10
+        name: tone
+        type: string
+        mapping:
+          - dps_val: "1"
+            value: White noise
+          - dps_val: "2"
+            value: Pink noise
+          - dps_val: "3"
+            value: Brown noise
+          - dps_val: "4"
+            value: Wall fan
+          - dps_val: "5"
+            value: Fan
+          - dps_val: "6"
+            value: Floor fan
+          - dps_val: "7"
+            value: Wind
+          - dps_val: "8"
+            value: Rain on roof
+          - dps_val: "9"
+            value: Rain
+          - dps_val: "10"
+            value: Thunderstorm
+          - dps_val: "11"
+            value: Light rain
+          - dps_val: "12"
+            value: Ocean wave
+          - dps_val: "13"
+            value: Water stream
+          - dps_val: "14"
+            value: Birds
+          - dps_val: "15"
+            value: Crickets
+          - dps_val: "16"
+            value: Frog
+          - dps_val: "17"
+            value: Fireplace
+          - dps_val: "18"
+            value: Campfire
+          - dps_val: "19"
+            value: Womb
+          - dps_val: "20"
+            value: Shushing
+          - dps_val: "21"
+            value: Cafe
+          - dps_val: "22"
+            value: Maracas
+          - dps_val: "23"
+            value: Clock
+          - dps_val: "24"
+            value: Buddha
+          - dps_val: "25"
+            value: Wind chimes
+          - dps_val: "26"
+            value: Steamship
+          - dps_val: "27"
+            value: Aircraft
+          - dps_val: "28"
+            value: Train
+          - dps_val: "29"
+            value: Lullaby
+          - dps_val: "30"
+            value: Music box
+          - dps_val: "31"
+            value: Twinkle little star
+          - dps_val: "32"
+            value: Good morning
+          - dps_val: "33"
+            value: Fantasy
+          - dps_val: "34"
+            value: Joyful
+      - id: 20
+        name: duration
+        type: integer
+        unit: "min"
+        range:
+          min: 0
+          max: 90
+        mapping:
+          - step: 30
+  - entity: select
+    name: Scene
+    icon: "mdi:palette"
+    category: config
+    dps:
+      - id: 12
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: wake_up
+            value: Wake up
+          - dps_val: help_sleep
+            value: Help sleep
+          - dps_val: breath
+            value: Breathing
+          - dps_val: colourful
+            value: Colourful
+          - dps_val: play
+            value: Play
+          - dps_val: reading
+            value: Reading
+          - dps_val: night
+            value: Nightlight
+          - dps_val: null
+            value: None
+  - entity: number
```

Comment:
I realise siren entities do not have a good UI in Home Assistant, but I don't think it is the place of this integration to work around that.  Instead of duplicating everything here, you can use input helpers to build a UI for the parameters to pass the siren.turn_on service call (and push the HA frontend developers to make a better UI for sirens).

Reply:
I see both sides here. While there should be a way to control the various settings from the device entry without needing to add extra helpers and build a dashboard, I see your point. Once changes are merged I still may wind up forking and use my own fork that way I could have these settings in the device page of my fork.
I think what we really need to do is build in functionality to support media_player component instead of using the siren component. I'll open a separate issue to address that, and if/when that is ever built, we could revisit this logic.